File: PostsAndComments/app/src/main/java/com/example/postscommentsapp/di/RepositoryModule.java
package com.example.postscommentsapp.di;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;

@Module
@InstallIn(SingletonComponent.class)
public class RepositoryModule {

    @Provides
    @Singleton
    public PostsRepository providePostsRepository(PostsRepositoryImpl repo) {
        return repo;
    }

    @Provides
    @Singleton
    public CommentsRepository provideCommentsRepository(CommentsRepositoryImpl repo) {
        return repo;
    }
}

File: PostsAndComments/app/src/main/java/com/example/postscommentsapp/di/RepositoryModule.kt
package com.example.postscommentsapp.di

import com.example.postscommentsapp.domain.repository.CommentsRepository
import com.example.postscommentsapp.domain.repository.PostsRepository
import com.example.postscommentsapp.data.repository.CommentsRepositoryImpl
import com.example.postscommentsapp.data.repository.PostsRepositoryImpl
import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
abstract class RepositoryModule {

    @Binds
    @Singleton
    abstract fun bindPostsRepository(
        impl: PostsRepositoryImpl
    ): PostsRepository

    @Binds
    @Singleton
    abstract fun bindCommentsRepository(
        impl: CommentsRepositoryImpl
    ): CommentsRepository
}
